### DISCUSSION TITLE
Grouping Or Accordioning Of Cells

### DISCUSSION THREAD

**Discussion by rdick** [nick-keller/react-datasheet-grid#58] [Q&A] [answered]
Hello,

I wanted to create an accordion within react-datasheet-grid. Therefore when the user clicks on a button it would open an area below that I could just insert subcomponent information there.

eg. you have you list of items => then in each you have details about that item when you click the accordion drop down

Is this possible?

Below is an example of something it might be like but it can just be a regular accordion as well with information in it but it would span the length of the table.
![image](https://user-images.githubusercontent.com/62073529/135357392-6dd3a8c7-f345-4a7c-8692-ada7a84c71c8.png)

**Comment by nick-keller** [comment]
Hello @rdick ,
This is completely feasable, but you need a little bit of work. Here are a few ideas for you, happy to go further if needed:
- You will need to manage the state of each group (opened or not) outside of DSG.
- Based on that state you can construct the rows you want to pass to DSG (eg. [group1,item1a,item1b,group2,item2a...]). Only pass the items of a group if it is opened.
- The `component` of each column should check if the row is a group or an item, and render its value accordingly.

And that's about it ;)

**Comment by nick-keller** [comment, marked as answer]
Example available in the doc: https://react-datasheet-grid.netlify.app/docs/examples/collapsible-rows
